src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -541,9 +541,7 @@ private void CompileNamedType(NamedTypeSymbol containingType)
 
                     case SymbolKind.Field:
                         {
-                            var field = (FieldSymbol)member;
-                            var fieldSymbol = (field.TupleUnderlyingField ?? field) as SourceMemberFieldSymbol;
-                            if ((object)fieldSymbol != null)
+                            if (member is FieldSymbol fieldSymbol)
```

Comment:
>member is FieldSymbol fieldSymbol [](start = 32, length = 33)

Is this going to emit code that is doing more than just an explicit cast? Given the kind check, we know for sure this is a FieldSymbol. The conditional logic is redundant, consider getting rid of it.

Reply:
BTW, I think the code would be more robust if we skip ```TupleErrorFieldSymbol``` here. And we did that before this change.

---
In reply to: [426203508](https://github.com/dotnet/roslyn/pull/44231#discussion_r426203508) [](ancestors = 426203508)